contracts/StaticOracle.sol

```diff
@@ -0,0 +1,221 @@
+        uint definedPools,
+        uint32 period
+    ) public view returns (uint256 quoteAmount) {
+        require(definedPools > 0, 'No defined pools');
+
+        OracleLibrary.WeightedTickData[] memory tickData = new OracleLibrary.WeightedTickData[](definedPools);
+        for (uint256 i; i < definedPools; i++) {
+            (tickData[i].tick, tickData[i].weight) = period > 0
+                ? OracleLibrary.consult(pools[i], period)
+                : OracleLibrary.getBlockStartingTickAndLiquidity(pools[i]);
+        }
+
+        int24 weightedTick = OracleLibrary.getWeightedArithmeticMeanTick(tickData);
+        return OracleLibrary.getQuoteAtTick(weightedTick, baseAmount, baseToken, quoteToken);
+    }    
+
+    /// @notice Takes a pair and a time period, and returns all pools that could be queried for that period
+    /// @param tokenA One of the pair's tokens
+    /// @param tokenB The other of the pair's tokens
+    /// @param period The period that we want to query for
+    /// @return queryablePools All pools that can be queried
+    /// @return amount How many of the elements in `queryablePools` are actually defined
+    function getQueryablePoolsForTiers(         
+        address tokenA,
+        address tokenB,
+        uint32 period
+    ) private view returns (address[] memory queryablePools, uint amount) {        
+        (address[] memory existingPools, uint definedPools) = getPoolsForTiers(tokenA, tokenB, knownFeeTiers);
+        queryablePools = new address[](definedPools);
+        for (uint i; i < definedPools; i++) {
+            if (OracleLibrary.getOldestObservationSecondsAgo(existingPools[i]) >= period) {
+                queryablePools[amount++] = existingPools[i];
+            }
+        }
+    }
+    
+    /// @notice Takes a pair and some fee tiers, and returns all pools that match those tiers
+    /// @param tokenA One of the pair's tokens
+    /// @param tokenB The other of the pair's tokens
+    /// @param feeTiers The fee tiers to consider when searching for the pair's pools
+    /// @return pools The pools for the given pair and fee tiers
+    /// @return definedPools How many of the elements in `pools` are actually defined
+    function getPoolsForTiers(address tokenA, address tokenB, uint24[] memory feeTiers) private view returns (address[] memory pools, uint definedPools) {        
+        pools = new address[](feeTiers.length);
+        for (uint i; i < feeTiers.length; i++) {
+            address pool = getPoolForTier(tokenA, tokenB, feeTiers[i]);
+            if (pool != address(0)) {
+                pools[definedPools++] = pool;
+            }
+        }
+    }    
+
+    /// @dev Returns the pool for the given token pair and fee. The pool contract may or may not exist
+    function getPoolForTier(address tokenA, address tokenB, uint24 feeTier) private view returns (address) {
+        return PoolAddress.computeAddress(factory, PoolAddress.getPoolKey(tokenA, tokenB, feeTier));
+    }
```

Comment:
Since `getPoolForTier`, is it necessary to split it? I don't recall gas consumption on internal call tho ... Just a thought.

Reply:
You are right, there is no need for a new function